### PR TITLE
Adds recipe for Tabulate project

### PR DESCRIPTION
**Pull request by fpelliccioni**
Specify library name and version:  **tabulate/1.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/tabulate/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import os
+from conans import ConanFile, tools
+
+class Tabulate(ConanFile):
+    name = "tabulate"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/p-ranav/tabulate"
+    description = "Table Maker for Modern C++"
+    topics = ("conan", "cpp17", "tabulate")
+    license = "MIT License"
```
Comment (uilianries) [inline recipes/tabulate/all/conanfile.py:10]:
```suggestion
    license = "MIT"
```

File: recipes/tabulate/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import os
+from conans import ConanFile, tools
+
+class Tabulate(ConanFile):
+    name = "tabulate"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/p-ranav/tabulate"
+    description = "Table Maker for Modern C++"
+    topics = ("conan", "cpp17", "tabulate")
```
Comment (uilianries) [inline recipes/tabulate/all/conanfile.py:9]:
```suggestion
    topics = ("header-only", "cpp17", "tabulate", "table", "cli")
```

File: recipes/tabulate/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
```
Comment (uilianries) [inline recipes/tabulate/all/conanfile.py:21]:
```suggestion

    def configure(self):
        compiler = str(self.settings.compiler)
        compiler_version = tools.Version(self.settings.compiler.version)

        if self.settings.compiler.cppstd:
            tools.check_min_cppstd(self, "17")
        else:
            self.output.warn("%s recipe lacks information about the %s compiler"
                             " standard version support" % (self.name, compiler))
        
        minimal_version = {
            "Visual Studio": "16",
            "gcc": "7.3",
            "clang": "6",
            "apple-clang": "10.0"
        }

        if compiler not in minimal_version:            
            self.output.info("%s requires a compiler that supports at least"
                             " C++17" % self.name)
            return

        if compiler_version < minimal_version[compiler]:
            raise ConanInvalidConfiguration("%s requires a compiler that supports"
                                            " at least C++17. %s %s is not" 
                                            " supported." % (self.name, compiler, Version(self.settings.compiler.version)))

```

File: recipes/tabulate/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import os
+from conans import ConanFile, tools
+
+class Tabulate(ConanFile):
+    name = "tabulate"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/p-ranav/tabulate"
+    description = "Table Maker for Modern C++"
```
Comment (uilianries) [inline recipes/tabulate/all/conanfile.py:8]:
```suggestion
    description = "Table Maker for Modern C++"
    settings = "compiler"
```

File: recipes/tabulate/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import os
+from conans import ConanFile, tools
```
Comment (uilianries) [inline recipes/tabulate/all/conanfile.py:2]:
```suggestion
from conans import ConanFile, tools
from conans.tools import Version
```

**Comment by fpelliccioni** [comment]
> @fpelliccioni Gracias por tu apoyo, hermano!

No es nada amigo!
Gracias a ustedes, Conan es una herramienta que me cambió la vida!
Gracias!

**Pull request merged**